circuit-prover/src/air/witness_air.rs

```diff
@@ -1,3 +1,26 @@
+//! [`WitnessAir`] is the central AIR in our construction of a recursive verifier. It represents the **global witness bus**,
+//! storing all witness indices and associated values used within the verifier circuit.
+//! The generic parameter `D` allows the AIR to handle values from an extension field of degree `D` over the base field.
+//!
+//! # Columns
+//!
+//! [`WitnessAir`] has `D + 1` columns:
+//!
+//! - 1 column for the index of the element,
+//! - `D` columns to store the value, where `D` is the degree extension of the used field compared to the current field
+//!
+//! Each row in the table increments the index by 1, starting from 0 (and so the index column is the range from 0 to `height - 1`).
+//!
+//!  # Constraints
+//!  - in the first row: ensure that `index = 0`.
+//!  - for transitions: `index_next - index_current - 1`.
+//!
+//! # Global Interactions
+//!
+//! Since this AIR serves as a witness bus, where the other chips read values from, it has interactions with all the other
+//! (primitive operation) AIRs.
```

Comment:
```suggestion
//! Since this AIR serves as a witness bus, where the other chips read values from, it has interactions with all the other
//! AIRs.
```

I believe non-primitive operations may also communicate with the Witness bus?

Reply:
I was thinking that nonprimitive operations *may* not have interactions with it but primitive operations always do. But I think you're right that it would be easier to just say it has interactions with everything.